akka-stream/src/main/boilerplate/akka/stream/scaladsl/SourceApply.scala.template

```diff
@@ -49,4 +49,22 @@ trait SourceApply {
   
   ]
 
+  [2..#/**
+   * Combines several sources with fun-in strategy like `Merge` or `Concat` and returns`Source`.
+   */
+  def combine[T, U]([#input1: Source[T, Any]#])(strategy: Int => Graph[UniformFanInShape[T, U], Unit]): Source[U, Unit] =
+    Source.wrap(FlowGraph.partial([#input1#])(([#M1#]) => ()) { implicit b ⇒
+      ([#s1#]) ⇒
+        import FlowGraph.Implicits._
+        val c = b.add(strategy(1))
+
+        [#s1 ~> c.in(0)#
+        ]
+
+        SourceShape(c.out)
+    })
+  #
```

Comment:
I wonder if using templating for this is the optimal solution (given the code duplication and scaling),
wdyt about something like the following?

~~~scala

 def combine[T, U](first: Source[T, _], second: Source[T, _], rest: Source[T, _]*)(strategy: Int ⇒ Graph[UniformFanInShape[T, U], Unit]): Source[U, Unit] =
    Source.wrap(FlowGraph.partial() { implicit b ⇒
      import FlowGraph.Implicits._

      val c = b.add(strategy(rest.size + 2))

      @tailrec def addRest(idx: Int, i: Iterator[Source[T, _]]): SourceShape[U] =
        if (i.hasNext) {
          i.next() ~> c.in(idx)
          addRest(idx + 1, i)
        } else SourceShape(c.out)

      first ~> c.in(0)
      second ~> c.in(1)
      addRest(2, rest.iterator)
    })

~~~